### ISSUE TITLE
Cannot define plugin properties on Android

### ISSUE THREAD

**Issue by rosshambrick** [ReactiveX/RxJava#2835]
The property key max length on Android devices/emulators is 31 characters thereby preventing the ability to set properties for use by the plug in system.  

The property key template is 29 characters on its own.
`rxjava.plugin.[class-name-here].implementation`

Is it likely that a shorter key could be used to enable this feature on Android?

**Comment by akarnokd** [comment]
I'm not familiar with the constraints on Android. How about we give you the following system property that wraps other system properties:

```
rxjava.props.1.key=rxjava.plugin.[class-name-here].implementation
rxjava.props.1.value=come.somepath.SomeClass
```

i.e., is it possible the use System.setProperty(key, value) on Android with long keys? Does the 31 character limit apply to command-line only?

**Comment by benjchristensen** [comment]
I have to say, that is a ridiculous limitation.

We can't change existing keys, but if someone wants to submit a PR with aliases that set both, that could work.

**Comment by akarnokd** [comment]
I've posted PR #3513 with a shorter scheme proposal.

**Comment by akarnokd** [comment]
Solution merged and delivered in 1.1.0
